historical_system_profiles/archiver.py

```diff
@@ -81,6 +83,36 @@ def _archive_profile(data, ptc, logger):
             "wrote %s to historical database (inv id: %s, captured_on: %s)"
             % (hsp.id, hsp.inventory_id, hsp.captured_on)
         )
+    # After the new hsp is saved, we need to check for any reason to alert Notifications
+    _check_need_for_notifications(host["id"])
+
+
+def _check_need_for_notifications(inventory_id)
```

Comment:
I am not sure about naming - we are not only checking for the need to send the notification we are also emitting the message if the need is there...

What about `_check_and_send_notifications` or something similar?